### PR TITLE
feat: add disallowScopes option

### LINKED ISSUE #176: Add config to disallow certain scopes
I'm the author of a public npm package called [cross-fetch](https://github.com/lquixada/cross-fetch). Every time a push is made on the `main` branch, the CI workflow is triggered. If the commit starts with `chore(release)`, this is the signal for CI to publish a release on npm and Github after all building, testing and linting is done. However if someone opens a PR with a title that starts with `chore(release)`, I'm not seeing a way to avoid the release. Would the maintainers consider adding a `disallowedScopes` config so users could avoid scopes such as `release`?

### PR DESCRIPTION
**Pull request by jmosawy**
Resolves #176

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,6 +55,10 @@ The action works without configuration, however you can provide options for cust
             ui
           # Configure that a scope must always be provided.
           requireScope: true
+          # Configure which scopes are disallowed in a PR title.
+          # For instance, by setting the value below, `chore(release): ...` or `ci(e2e,release): ...` will be rejected
```
Comment (amannn) [inline README.md:59]: Added a dot at the end and wrapped the lines a bit nicer:

```suggestion
          # Configure which scopes are disallowed in PR titles. For instance,  by setting
          # the value below, `chore(release): ...` or `ci(e2e,release): ...` will be rejected.
```

File: action.yml [hunk 2]
```diff
@@ -17,6 +17,9 @@ inputs:
   requireScope:
     description: "Configure that a scope must always be provided."
     required: false
+  disallowScopes:
+    description: 'Configure a list of disallowed scopes to be used on a PR.'
```
Comment (amannn) [inline action.yml:21]: Synced over the description from the README:

```suggestion
    description: 'Configure which scopes are disallowed in PR titles.'
```

File: src/validatePrTitle.js [hunk 3]
```diff
@@ -89,6 +95,15 @@ module.exports = async function validatePrTitle(
     );
   }
 
+  const disallowedScopes = givenScopes
+    ? givenScopes.filter(isDisallowedScope)
+    : [];
+  if (disallowScopes && disallowedScopes.length > 0) {
+    throw new Error(
+      `Disallowed scope(s) are found: ${disallowScopes.join(', ')}`
```
Comment (amannn) [inline src/validatePrTitle.js:103]: We pluralize in other error messages as well, I think we should do here too:

```suggestion
      `Disallowed ${disallowedScopes.length === 1 ? 'scope was' : 'scopes were'} found: ${disallowScopes.join(', ')}`
```

File: src/validatePrTitle.test.js [hunk 4]
```diff
@@ -98,6 +98,68 @@ describe('defined scopes', () => {
...
+      });
+
+      it('throws when multiple scopes are provided and more than one of them are present in disallowScopes', async () => {
+        await expect(
+          validatePrTitle('fix(release,test): Bar', {
+            disallowScopes: ['release', 'test']
+          })
+        ).rejects.toThrow('Disallowed scope(s) are found: release, test');
+      });
+    });
```
Comment (amannn) [inline src/validatePrTitle.test.js:161]: Thanks for adding such an exhaustive test suite! 👏

**Review by amannn — COMMENTED** [review]
Looks really good to me, thanks for proposing this PR!

The contributors guide has information on how to test this on CI: https://github.com/amannn/action-semantic-pull-request/blob/master/CONTRIBUTORS.md.

**Comment by jmosawy** [comment]
> Looks really good to me, thanks for proposing this PR!
> 
> The contributors guide has information on how to test this on CI: https://github.com/amannn/action-semantic-pull-request/blob/master/CONTRIBUTORS.md.

Thank you @amannn for the valuable feedback. I managed to fix them all now. Apologies for making a little bit of a mess here, but hopefully next time everything would look much neater. 😄 

I have a PR open in my forked version [here](https://github.com/jmosawy/action-semantic-pull-request/pull/1). You can check this CI log [here](https://github.com/jmosawy/action-semantic-pull-request/runs/6281304831?check_suite_focus=true) that shows the failed test because of including the `release` scope.

Let me know if there's anything else I should take care of.

**Review by amannn — APPROVED** [review]
Great work! 👏 Thank you so much for your valuable contribution @jmosawy!

**Pull request merged**